In [3]:
import urllib.request
import json

# Request data goes here
# The example below assumes JSON formatting which may be updated
# depending on the format your endpoint expects.
# More information can be found here:
# https://docs.microsoft.com/azure/machine-learning/how-to-deploy-advanced-entry-script
data = {
  "Inputs": {
    "WebServiceInput0": [
      {
        "symboling": 2,
        "normalized-losses": 150,
        "make": "toyota",
        "fuel-type": "gas",
        "aspiration": "std",
        "num-of-doors": "four",
        "body-style": "sedan",
        "drive-wheels": "fwd",
        "engine-location": "front",
        "wheel-base": 95.7,
        "length": 165.3,
        "width": 64.4,
        "height": 53.0,
        "curb-weight": 2145,
        "engine-type": "ohc",
        "num-of-cylinders": "four",
        "engine-size": 97,
        "fuel-system": "2bbl",
        "bore": 3.15,
        "stroke": 3.03,
        "compression-ratio": 9.0,
        "horsepower": 69,
        "peak-rpm": 4900,
        "city-mpg": 31,
        "highway-mpg": 37,
        "price": 15000
      }
    ]
  },
  "GlobalParameters": {}
}




body = str.encode(json.dumps(data))

url = 'URL removed for security reasons'
# Replace this with the primary/secondary key, AMLToken, or Microsoft Entra ID token for the endpoint
api_key = 'API key removed for security reasons'
if not api_key:
    raise Exception("A key should be provided to invoke the endpoint")


headers = {'Content-Type':'application/json', 'Accept': 'application/json', 'Authorization':('Bearer '+ api_key)}

req = urllib.request.Request(url, body, headers)

try:
    response = urllib.request.urlopen(req)

    result = response.read()
    print(result)
except urllib.error.HTTPError as error:
    print("The request failed with status code: " + str(error.code))

    # Print the headers - they include the requert ID and the timestamp, which are useful for debugging the failure
    print(error.info())
    print(error.read().decode("utf8", 'ignore'))


b'{"Results": {"WebServiceOutput0": [{"symboling": 2, "make": "toyota", "fuel-type": "gas", "aspiration": "std", "num-of-doors": "four", "body-style": "sedan", "drive-wheels": "fwd", "engine-location": "front", "wheel-base": 95.7, "length": 165.3, "width": 64.4, "height": 53.0, "curb-weight": 2145, "engine-type": "ohc", "num-of-cylinders": "four", "engine-size": 97, "fuel-system": "2bbl", "bore": 3.15, "stroke": 3.03, "compression-ratio": 9.0, "horsepower": 69.0, "peak-rpm": 4900.0, "city-mpg": 31, "highway-mpg": 37, "price": 15000.0, "Scored Labels": 8296.392102109232}]}}'


This script sends a sample JSON payload to the deployed Azure Machine Learning endpoint to validate that the model scoring API is functioning correctly.
For security reasons, the actual endpoint URL and API key have been removed from the script. Users should configure these values through environment variables before executing the code.

A sample response from a successful request (with sensitive information removed) is shown below, demonstrating that the endpoint was tested and is working as expected:

b'{"Results": {"WebServiceOutput0": [..., "Scored Labels": 8296.39]}}'
This confirms that the model is correctly returning predictions.